### DISCUSSION TITLE
Create cube using narrow dataset

### DISCUSSION THREAD

**Discussion by kascp** [atoti/atoti#811] [General]
Hi
Do you see any problems with creating the cube with a narrow (long) dataset instead of a wide dataset?
All the tutorials I looked at are using a wide dataset and ending up having multiple measures.
I plan to build the cube with a narrow dataset. This means I end up only with one measure AMOUNT.SUM.
Thanks
Philipp

wide
<html>

book | instrument | delta usd | gamma usd | vega usd
-- | -- | -- | -- | --
A | 1 | 100 | 10 | 20
B | 2 | 50 | 5 | 10
</html>

narrow
<html> 

book | instrument | amount type | amount 
-- | -- | -- | -- 
A | 1 | delta usd | 100  
A | 1 | gamma usd | 10 
A | 1 | vega usd | 20  
B | 2 | delta usd | 50  
B | 2 | gamma usd | 5 
B | 2 | vega usd | 10 

</html>

**Comment by mailys-ds** [comment]
Hi Philipp, 

It's not an issue at all to manipulate narrow datasets, with only few measures, Indeed, if Atoti can handle multiple measures, it will be able to handle one ;).

NB: Also with the narrow dataset you gave as example, if you create a cube with the mode _auto_ and if your column _instrument_ is non-numeric, you will end up with 3 measures: amount.SUM, amount.MEAN and contributors.COUNT.

**Comment by kascp** [comment]
Hi!
Thanks for the swift reply. Was just wondering if there is any big disavantage with using a narrow dataset when building the cubes.

Agree with your NB, I immediately kick out the MEAN because it does not have value for me.

Additional question:
When creating a tree table, I do the following - Rows = book, Columns = amount type, Measures = amount.SUM
The table looks nice, but the column headers look as follows
delta chf----------gamma chf--------vega chf
amount.SUM-----amount.SUM------amount.SUM

Given I use only one measure, is it possible to remove/hide the amount.SUM in the column name? It does not add value here.

Also, is it possible to sort the columns based on a defined ordering?

Appreciate your help
Thanks
Philipp
